### ISSUE TITLE
feat(android): Touch keyboard height adjustability

### ISSUE THREAD

**Issue by bjeenkhoorn** [keymanapp/keyman#3225]
Working with the touch keyboard for android is a problem because the keys are so small.  The height of the keys cannot be changed which is a big put-off for usability. Most people who will be using the touch keyboard live in West Africa and are farmers at least part of the year. Those men have big hands and now use android phones. 

Looking at our v 1.0 of the tough layout it would be nice to change the height of the keys. I know there is a setting for the width of the keys, but can the height also be adjusted ? Especially with the additional row, I think it needs more space.
![image](https://user-images.githubusercontent.com/47192048/84256853-c588e480-ab14-11ea-930b-7c29a912b9d8.png)

One solution could be, as a temporary measure perhaps is to reduce the size of the prediction window. This height is quite luxurious compared to what is attributed to the keys. If you could scrape off 20% of that and distribute it among the other keys that would give some air at least.

But what is really needed is the height to be adjustable, be it completely flexible or with "increments" of a certain added hight per row.

 - OS:  Android 9

**Comment by MakaraSok** [comment]
This feature is also requested by other users: https://community.software.sil.org/t/can-i-make-the-whole-keyboards-height-taller-touch-keyboard/4067.

**Comment by MattGyverLee** [comment]
> Notes: smaller devices will struggle with more screen real-estate taken over by a keyboard. I've seen custom keyboards that took over 80% of the screen, which made many apps unusable. Some earlier versions of devices had a fixed area that could be used by the keyboard; most modern devices don't have that as far as I know, but it's still something we need to account for.

I have the opposite problem, a relatively high-resolution screen Pixel 3XL, a 4-row keyboard like @bjeenkhoorn , and buttons that even my tiny fingers can hardly handle.

I hoping for a slider that would adjust the "automagic" height up or down (within reasonable limits) on my app's configuration page.  I'd be very happy even if it was a "use at your own risk" feature. I've been trying (without sucess) to figure out a way to increase row height in the code.

**Comment by mcdurdin** [comment]
Unfortunately, I don't think this issue will make it into 14.0 but it is a priority for 15.0, along with other on screen keyboard style and display improvements.

**Comment by darcywong00** [comment]
We'll try to have a design phase for this in 15.0 

Note, some other users request the OSK to be smaller (I assume landscape orientation?)
(4th point on https://community.software.sil.org/t/better-keyman-for-andriod/4791)

**Comment by darcywong00** [comment]
A draft in progress of the design spec on Keyman for Android
https://docs.google.com/document/d/1B6SRPbKGe06vM8j_-cyAGpKrMmINYe-slJq9UIPlYws/edit

**Comment by jahorton** [comment]
Ah, right.  This is a combo issue for both Android and iOS.  I don't think I'll be able to approach the iOS portion until at least A15S8.

**Comment by darcywong00** [comment]
On Keyman for Android, here's a rough mockup of the upcoming new menu for adjusting the keyboard height.
Resizing the "image" will adjust the keyboard height for the current orientation (portrait vs landscape). The scope will also not include adjusting the suggestion banner height.

![adjust keyboard height mockup](https://user-images.githubusercontent.com/7358010/129850081-581a9ef2-a31f-4ec4-9773-85f47e49c8ba.png)

**Comment by mcdurdin** [comment]
Is this resolved by #5606?

**Comment by darcywong00** [comment]
> Is this resolved by #5606?

It's still open for iOS

**Comment by MattGyverLee** [comment]
Is this feature available in KMEA? How would I trigger it?

**Comment by darcywong00** [comment]
> Is this feature available in KMEA? How would I trigger it?

@MattGyverLee - I was originally going to say it's only available in the Keyman for Android 15.0 app. But it turns out there's a new API in Keyman Engine for Android 15.0 you can try:

```
  public static void applyKeyboardHeight(Context context, int height) {
```

where `height` units is dp. see: https://github.com/keymanapp/keyman/blob/892416df8f096285357ef9df9467be1519da48ab/android/KMEA/app/src/main/res/values/dimens.xml#L5

**Comment by mcdurdin** [comment]
Splitting iOS into separate issue #5897.

Removing web tag for now as web platform would be lower priority for now.

(@darcywong00, workflow note: if an issue is partially addressed, then we should split the *unfinished* components into a new issue, rather than removing the finished components -- otherwise it's hard to track what is finished and when.)
